membrane_curvature/tests/test_mdakit_membcurv.py

```diff
@@ -144,44 +151,152 @@ def test_mean_curvature_all():
-    selection = u.select_atoms('index 0:3')
-    core_fast_leaflet(u, z_calc, n_cells, selection, max_width)
-    for z, z_test in zip(MEMBRANE_CURVATURE_DATA['grid']['small']['upper'], z_calc):
-        print(z, z_test)
-        assert_almost_equal(z, z_test)
-
-
-@pytest.mark.parametrize('factor', [1, 2])
-@pytest.mark.parametrize('dummy_coord', [
-    # dummy coordinates (x,y) in grid of 3x3
-    (0, 0), (1, 0), (2, 0),
-    (0, 1), (1, 1), (2, 1),
-    (0, 2), (1, 2), (2, 2),
-    # dummy coordinates (x,y) in grid of 5x5
-    (0, 0), (1, 0), (2, 0), (3, 0), (4, 0),
-    (0, 1), (1, 1), (2, 1), (3, 1), (4, 1),
-    (0, 2), (1, 2), (2, 2), (3, 2), (4, 2),
-    (0, 3), (1, 3), (2, 3), (3, 3), (4, 3),
-    (0, 4), (1, 4), (2, 4), (3, 4), (4, 4)
+@pytest.mark.parametrize('n_cells, grid_z_coords', [(
+    # number of cells
+    3,
+    # z position in grid
+    np.array([[10., 10., 10.],
+              [10., 10., 10.],
+              [10., 10., 10.]])
+)])
+def test_avg_unit_cell_identity(n_cells, grid_z_coords):
+    z_ref = np.zeros([n_cells, n_cells])
+    # number of beads per grid
+    unit = np.ones([n_cells, n_cells])
+    unit_cell = avg_unit_cell(z_ref, grid_z_coords, unit)
+    for z_position, cell in zip(unit_cell, range(n_cells)):
+        assert z_position[cell] == 10.
+
+
+@pytest.mark.parametrize('n_cells, grid_z_coords', [(
+    # number of cells
+    3,
+    # z position in grid
+    np.array([[10., 20., 30.],
+              [10., 20., 30.],
+              [10., 20., 30.]]
+             ))])
+def test_avg_unit_cell_identity_other_values(n_cells, grid_z_coords):
+    z_ref = np.zeros([n_cells, n_cells])
+    unit = np.ones([n_cells, n_cells])
+    z_avg = avg_unit_cell(z_ref, grid_z_coords, unit)
+    for cell in range(n_cells):
+        assert_almost_equal(z_avg[cell], grid_z_coords[cell])
+
+
+@pytest.mark.parametrize('n_cells, grid_z_coords, grid_norm, grid_avg', [(
+    # number of cells
+    3,
+    # sum of z coordinate in grid
+    np.array([[10., 10., 10.],
+              [10., 10., 10.],
+              [10., 10., 10.]]),
+    # grid number of beads per unit
+    np.array([[2., 1., 1.],
+              [1., 2., 1.],
+              [1., 1., 2.]]),
+    # avg z coordinate in grid
+    np.array([[5., 10., 10.],
+              [10., 5., 10.],
+              [10., 10., 5.]]),
+)])
+def test_avg_unit_cell_more_beads(n_cells, grid_z_coords, grid_norm, grid_avg):
+    z_ref = np.zeros([n_cells, n_cells])
+    unit_cell = avg_unit_cell(z_ref, grid_z_coords, grid_norm)
+    for cell in range(n_cells):
+        assert_almost_equal(unit_cell[cell], grid_avg[cell])
+
+
+@pytest.mark.parametrize('n_cells, max_width, z_avg', [(
+    # number of cells
+    # max_width is 3nm/30A (from GRO_PO4_SMALL)
+    3, 30.,
+    # z ref of surface
+    # z coordinate in grid
+    np.array(([150., 150., 120.],
+              [150., 120., 120.],
+              [150., 120., 120.])))])
+def test_derive_surface(small_grofile, n_cells, max_width, z_avg):
```

Comment:
Could you please rewrite this test to not use `parametrize` (since it does not use multiple test cases) and only with `assert_almost_equal` instead of the for loop?